src/audio/kpb.c

```diff
@@ -1113,9 +1113,9 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 							   host_period_size / bytes_per_ms) /
 					 KPB_DRAIN_NUM_OF_PPL_PERIODS_AT_ONCE;
 			period_bytes_limit = host_period_size;
-			comp_info(dev, "kpb_init_draining(): sync_draining_mode selected with interval %d [uS].",
-				  drain_interval * 1000 /
-				  clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1));
+			comp_info(dev, "kpb_init_draining(): sync_draining_mode selected with interval %u [uS].",
+				  (unsigned int)(drain_interval * 1000 /
+						 clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1)));
 		} else {
```

Comment:
@lyakh I'm also having some warnings, on trace, when building SOF with Zephyr:
```
In file included from sof/sof/zephyr/../src/include/sof/lib/dai.h:27,
                 from sof/sof/src/lib/dai.c:7:
sof/sof/src/lib/dai.c: In function 'dai_group_get':
sof/sof/zephyr/include/sof/trace/trace.h:35:4: warning: implicit declaration of function 'printk'; did you mean 'cbprintf'? [-Wimplicit-function-declaration]
   35 |    printk("%llu: " format "\n", platform_timer_get(NULL), \
      |    ^~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:287:2: note: in expansion of macro '_log_message'
  287 |  _log_message(trace_log_filtered, true, LOG_LEVEL_CRITICAL, class, ctx, id_1, \
      |  ^~~~~~~~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:290:2: note: in expansion of macro '_trace_error_with_ids'
  290 |  _trace_error_with_ids(class, ctx, id_1, id_2, format, ##__VA_ARGS__)
      |  ^~~~~~~~~~~~~~~~~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:374:2: note: in expansion of macro 'trace_error_with_ids'
  374 |  trace_error_with_ids(_TRACE_INV_CLASS, ctx, \
      |  ^~~~~~~~~~~~~~~~~~~~
sof/sof/src/lib/dai.c:85:3: note: in expansion of macro 'tr_err'
   85 |   tr_err(&dai_tr, "dai_group_get(): invalid group_id %u",
      |   ^~~~~~
```
Any idea how can I get rid of them?
Do I need to do some LOGGING configuration for Zephyr?
Thanks!

Reply:
@iuliana-prodan in fact that looks like a bug in zephyr/include/sof/trace/trace.h to me. It should include sys/printk.h under `#ifdef __ZEPHYR__`, would you like to submit a fix?

Reply:
@lyakh ok, sure, I can do it.
Let me check it and I'll submit the fix.
Thanks!